### PR TITLE
[nginx] Add support for TCP/UDP ports

### PR DESCRIPTION
**Pull request by goruha**
## What
* Provide env vars to proxy `tcp` and `udp`

## Why
* Expose non-http services via ingress

### CODE REVIEW

File: helmfile.d/0320.nginx-ingress.yaml [hunk 1]
```diff
@@ -60,6 +60,17 @@ releases:
           enabled: {{ env "NGINX_INGRESS_METRICS_ENABLED" | default "false" }}
         metrics:
           enabled: {{ env "NGINX_INGRESS_METRICS_ENABLED" | default "false" }}
+      ### Optional: NGINX_INGRESS_TCP; e.g. 8080:default/example-tcp-svc:9000,8081:staging/example-tcp-svc:9000
+      tcp:
+      {{- range env "NGINX_INGRESS_TCP" | splitList "," }}
+        {{ regexReplaceAll "(\\d*):(.*)" . "$1: $2" }}
```
Comment (aknysh) [inline helmfile.d/0320.nginx-ingress.yaml]: can you explain in comments what the regex does and why?

**Review by aknysh — COMMENTED** [review]
see comments

File: helmfile.d/0320.nginx-ingress.yaml [hunk 2]
```diff
@@ -60,6 +60,17 @@ releases:
...
+      {{- end }}
+      ### Optional: NGINX_INGRESS_UDP; e.g. 8080:default/example-udp-svc:9000,8081:staging/example-udp-svc:9000
+      udp:
+      {{- range env "NGINX_INGRESS_UDP" | splitList "," }}
+        {{ regexReplaceAll "(\\d*):(.*)" . "$1: $2" }}
```
Comment (osterman) [inline helmfile.d/0320.nginx-ingress.yaml]: Since `$2` contains a `:`, seems like it should be `| quote` ed.

**Review by osterman — CHANGES_REQUESTED** [review]
Agree with @aknysh; describe the regex in a simple `#` comment. Consider using `| quote`

File: helmfile.d/0320.nginx-ingress.yaml (hunk 1 above)
Comment (goruha) [reply helmfile.d/0320.nginx-ingress.yaml]: We need to map 
`8080:default/example-tcp-svc:9000,8081:staging/example-tcp-svc:9000`
to
```
8080: default/example-tcp-svc:9000
8081: staging/example-tcp-svc:9000
```

Comment (aknysh) [reply helmfile.d/0320.nginx-ingress.yaml]: can you describe the regex in a simple # comment?
what it does and why.

File: helmfile.d/0320.nginx-ingress.yaml (hunk 2 above)
Comment (goruha) [reply helmfile.d/0320.nginx-ingress.yaml]: nope. I tested - that's ok

**Review by osterman — APPROVED** [review]

**Pull request merged**